javascript/modules/client/scripts/generate-readme.js

```diff
@@ -0,0 +1,78 @@
+const fs = require("fs");
+const { argv } = require("process");
+
+generateReadme();
+
+async function generateReadme() {
+  const path = argv[2];
+  const outputFile = argv[3];
+  const mdRegex = /(\/\* MARKDOWN)(.|\n)*?(\*\/)/;
+  const files = getFileNames(path);
+  if (fs.existsSync(outputFile)) {
+    await fs.promises.rm(outputFile);
+  }
+  for (const file of files) {
+    fs.readFile(file, async (err, data) => {
+      if (err) throw err;
+      try {
+        const md = getText(
+          data.toString(),
+          mdRegex,
+          ["/* MARKDOWN", "*/"],
+          true
+        );
+        const code = data.toString().replace(mdRegex, "");
+        let readmeStr = md;
+        if (code && code !== "\n") {
+          readmeStr = md + `\n\`\`\`ts\n` + code + `\n\`\`\`\n`;
+        }
```

Comment:
Does this support the case of a file having (1) markdown (2) code (3) more markdown (4) more code ?

I would suggest breaking this into simpler, more focused functions (doesn't need to be perfect though):
- buildExampleFile(path) => string
- parseFile(path) => tokens: Array<{ type: "code" | "markdown", content: string}>
- processTokens(tokens) => string

Reply:
No, right now it does not support that, and with the aproach that we have rigth now we cannot support it, the only batch of stuff that has some kind of pattern of separator is the markdown
```ts
/* MARKDOWN
....
*/
We use the `mdRegex` to detect this pattern and get the section, and then we use the same `mdRegex` to delete this section and get the code. So if we have the pattern of having md -> code ->md -> code, we woun't be able to separate the firs section of code from the secon one and  generate an ordered markdown after that.
```